flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/codegen/ExpressionReducer.scala

```diff
@@ -94,8 +99,15 @@ class ExpressionReducer(config: TableConfig)
     val literalTypes = literals.map(e => FlinkTypeFactory.toTypeInfo(e.getType))
     val resultType = new RowTypeInfo(literalTypes: _*)
 
+    val parameters = if (config.getConfiguration != null) {
+      config.getConfiguration
```

Comment:
Currently, we use `ExecutionConfig#getJobParameter` as the UDF's configuration in runtime. However, I think we should use `TableConfig#getConfiguration`. I opened another issue FLINK-14514 to discuss this.